packages/plugins/ens/src/index.js

```diff
@@ -119,30 +134,32 @@ class ENS {
   }
 
   addArtifactFile(_params, cb) {
-    const config = this.getEnsConfig();
-    this.events.request("pipeline:register", {
-      path: [this.config.embarkConfig.generationDir, 'config'],
-      file: 'namesystem.json',
-      format: 'json',
-      content: Object.assign({}, this.embark.config.namesystemConfig, config)
-    }, cb);
+    this.getEnsConfig((err, config) => {
+      this.events.request("pipeline:register", {
+        path: [this.config.embarkConfig.generationDir, 'config'],
+        file: 'namesystem.json',
+        format: 'json',
+        content: Object.assign({}, this.embark.config.namesystemConfig, config)
+      }, cb);
+    });
   }
 
   async setProviderAndRegisterDomains(cb = (() => {})) {
-    const config = this.getEnsConfig();
-    if (this.doSetENSProvider) {
-      this.setupEmbarkJS(config);
-    }
+    this.getEnsConfig(async (err,config) => {
```

Comment:
If suggestion above is implemented, this could remove the callback and change to:
```
const config = await this.config;
```